torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -30,6 +30,51 @@
 namespace c10d {
 namespace detail {
 
+class timing_guard {
+  Counter& counter_;
+  typedef std::chrono::time_point<std::chrono::high_resolution_clock>
+      time_point;
+  time_point start_;
+
+ public:
+  timing_guard(Counter& counter)
+      : counter_(counter), start_(std::chrono::high_resolution_clock::now()) {}
+
+  ~timing_guard() {
+    stop();
+  }
+
+  void stop() {
+    if (start_ != time_point()) {
```

Comment:
is it valid to construct a timing_guard without initializing its start_?  Should this just assert start_ is valid instead of noop-ing?

Reply:
`start_ == time_point()` when `start_` is not initialized but I'm not sure if it is a defined behavior or compiler-dependent. Likely it is safe.

Reply:
It's not valid, but the ctor does initialize it.